nucypher-core-python/src/lib.rs

```diff
@@ -720,6 +733,71 @@ impl ThresholdDecryptionRequest {
     }
 }
 
+//
+// E2EEThresholdDecryptionRequest
+//
+#[pyclass(module = "nucypher_core")]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct E2EEThresholdDecryptionRequest {
+    backend: nucypher_core::E2EEThresholdDecryptionRequest,
+}
+
+#[pymethods]
+impl E2EEThresholdDecryptionRequest {
+    #[getter]
+    pub fn decryption_request(&self) -> ThresholdDecryptionRequest {
+        self.backend.decryption_request.clone().into()
+    }
+
+    #[getter]
+    pub fn response_encrypting_key(&self) -> PublicKey {
+        self.backend.response_encrypting_key.into()
+    }
+
+    #[staticmethod]
+    pub fn from_bytes(data: &[u8]) -> PyResult<Self> {
+        from_bytes::<_, nucypher_core::E2EEThresholdDecryptionRequest>(data)
+    }
+
+    fn __bytes__(&self) -> PyObject {
+        to_bytes(self)
+    }
+}
+
+//
+// EncryptedThresholdDecryptionRequest
+//
+
+#[pyclass(module = "nucypher_core")]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct EncryptedThresholdDecryptionRequest {
+    backend: nucypher_core::EncryptedThresholdDecryptionRequest,
+}
+
+#[pymethods]
+impl EncryptedThresholdDecryptionRequest {
+    #[getter]
+    pub fn id(&self) -> u16 {
+        self.backend.ritual_id
+    }
+
+    pub fn decrypt(&self, sk: &SecretKey) -> PyResult<E2EEThresholdDecryptionRequest> {
+        self.backend
+            .decrypt(sk.as_ref())
+            .map(E2EEThresholdDecryptionRequest::from)
+            .map_err(|err| PyValueError::new_err(format!("{}", err)))
```

Comment:
Unrelated to this PR, but maybe we could turn this often-repeated conversion into a helper function, similarly how we use `.map_err(map_js_err)` in WASM bindings:
```suggestion
            .map_err(mapy_py_err)
```